### ISSUE TITLE
Add Okio or KotlinxIO Path and File options & arguments

### ISSUE THREAD

**Issue by Wavesonics** [ajalt/clikt#568]
Currently Path and File option types are only for the JVM as they rely on those JVM classes.

But getting versions of these that would work on native targets would be great.

I looked at just creating them in my own project to get unblocked, but it looks like some of the types and functions use by those methods under the hood are marked internal I think.

**Comment by ajalt** [comment]
This is a duplicate of #492, closing in favor of the earlier issue. To summarize, if you don't need the extra functionality, you can do `option().convert{ it.toWhatever() }`
